core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -651,10 +651,15 @@ class WorkspaceService(protected val userInfo: UserInfo,
     }
   }
 
-  def getTags(query: Option[String], limit: Option[Int] = None): Future[Seq[WorkspaceTag]] =
-    dataSource.inTransaction { dataAccess =>
-      dataAccess.workspaceQuery.getTags(query, limit)
-    }
+  def getTags(query: Option[String], limit: Option[Int] = None): Future[Seq[WorkspaceTag]] = {
+    for {
+      workspacesForUser <- samDAO.getPoliciesForType(SamResourceTypeNames.workspace, userInfo)
+      workspaceIdsForUser = workspacesForUser.map(resource => UUID.fromString(resource.resourceId)).toSeq
```

Comment:
you might want to add a `Try` around the `UUID.fromString` - see also https://github.com/broadinstitute/rawls/blob/0cbb762a1ce1f4493ee9cae04b628861fab52447/core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala#L688.

At some point during corner-case testing, someone manually added non-UUID Sam resources of type "workspace". Any user with access to such a Sam resource would get a fatal error on `UUID.fromString` and this API will never complete.

Reply:
Ugh, that's right. Handled in 062a04380